### PR TITLE
docs(calling): improve samples UI. add back getDevice

### PR DESCRIPTION
**Pull request by rajeshtezu**
# COMPLETES #< [SPARK-458857](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-458857) >

## This pull request addresses

- Updated the samples page with improved UI

## by making the following changes

**Before**

Screenshot 2023-09-20 at 3 18 36 PM
Screenshot 2023-09-20 at 3 18 49 PM
Screenshot 2023-09-20 at 3 19 06 PM
Screenshot 2023-09-20 at 3 19 20 PM
Screenshot 2023-09-20 at 3 19 34 PM
Screenshot 2023-09-20 at 3 19 45 PM

**After**

screencapture-localhost-8000-samples-calling-2023-09-23-10_58_06

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [ ] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [ ] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: docs/samples/calling/style.css [hunk 1]
```diff
@@ -254,18 +254,13 @@ input[type=email]:focus {
...
-  grid-template-columns: 1fr 400px;
-  padding-bottom: 1rem;
+  overflow: auto;
+  height: 100vh;
 }
 
 .docs {
-  overflow: auto;
-  height: 98vh;
+  max-width: 1400px;
```
Comment (karthikraji2020) [inline docs/samples/calling/style.css:262]: can we convert this `px` value to ``rem value here ?
replace to
`max-width: 87.5rem`

File: docs/samples/calling/style.css [hunk 2]
```diff
@@ -329,3 +328,48 @@ select.ServiceData {
...
+
+.text-color {
+  color: #555;
+}
+
+input[type="string"],
+select {
+  padding: 0 0.5rem;
+  height: 1.75rem;
+  border-radius: 4px;
```
Comment (karthikraji2020) [inline docs/samples/calling/style.css:365]: can we use rem value here as well ?

File: docs/samples/calling/style.css [hunk 3]
```diff
@@ -16,18 +15,19 @@ video {
...
   font-size: 2.2rem;
+  margin: 0 0 1rem 0;
 }
 
 section {
   background-color: #fff;
   padding: 1rem;
-  margin-bottom: 2rem;
+  margin-bottom: 1.5rem;
+  border-radius: 8px;
```
Comment (karthikraji2020) [inline docs/samples/calling/style.css:25]: Here as well.

**Review by karthikraji2020 — COMMENTED** [review]
could you add a screenshot like before and after code changes format ?

File: docs/samples/calling/style.css (hunk 2 above)
Comment (rarajes2) [reply docs/samples/calling/style.css:365]: We should not change border radius if the root font-size changes.

File: docs/samples/calling/style.css (hunk 3 above)
Comment (rarajes2) [reply docs/samples/calling/style.css:25]: Same as above

File: docs/samples/calling/index.html [hunk 4]
```diff
@@ -9,495 +9,518 @@
...
-              </label>
-              <label>
-                <input type="checkbox" name="ts-media-direction" value="sendVideo" checked="">sendVideo
-              </label>
-            </div>
-          </div>
+              <!--- Transfer section -->
+              <div id="transfersection">
+                <fieldset>
+                  <legend>Call Transfer</legend>
```
Comment (Kesari3008) [inline docs/samples/calling/index.html:165]: We can just keep the Call Transfer title, we can remove the outer border associated with the title and bring it to the same indentation level as other buttons

File: docs/samples/calling/app.js [hunk 5]
```diff
@@ -95,6 +95,16 @@ const devicesById = {};
 const img = new Image();
 const vmAvatarImg = new Image(100);
 
+document.querySelectorAll('.collapsible').forEach((el) => {
```
Comment (Kesari3008) [inline docs/samples/calling/app.js:98]: Where are we using tghs ?

File: docs/samples/calling/app.js [hunk 6]
```diff
@@ -601,18 +610,17 @@ function clearMediaDeviceList() {
 }
 
 async function getMediaDevices() {
-  const cameras = await window.Media.Media.getCameras();
-
+  const cameras = await calling.Media.Media.getCameras();
```
Comment (Kesari3008) [inline docs/samples/calling/app.js:613]: Try accessing media directly from the calling client, we might not need to re-export it from index.ts and populate it inside calling instance. callingClient.mediaEngine.Media should suffice. Try this and update the change if it works

**Review by Kesari3008 — CHANGES_REQUESTED** [review]

File: docs/samples/calling/index.html [hunk 7]
```diff
@@ -9,495 +9,518 @@
...
-                Quality</button>
+          
+            <form id="call-create">
+              <fieldset>
+                <legend>Make a call</legend>
+                <div class="u-mv">
+                  <input id="destination" name="destination" placeholder="Destination" type="string">
+                  <p></p>
+                  <button id="create-call-action" type="submit" class="btn-code">Make Call</button>
+                  <button id="end-call" type="button" onclick="endCall()" class="btn-code">End Call</button>
```
Comment (Kesari3008) [inline docs/samples/calling/index.html:131]: maybe we can keep end call disabled here as well until call is established

File: packages/webex/src/calling.js [hunk 8]
```diff
@@ -28,6 +28,7 @@ class Calling extends EventEmitter {
     this.callingConfig = callingConfig;
     this.log = WebexCalling.Logger;
     this.log.setLogger(callingConfig.logger.level, CALLING_FILE);
+    this.Media = WebexCalling.Media;
```
[the remark this answers is not in the document]
Comment (rarajes2) [reply packages/webex/src/calling.js:31]: Done

File: docs/samples/calling/index.html [hunk 9]
```diff
@@ -9,495 +9,518 @@
...
+
+                  <pre id="call-object">Call-object</pre>
```
[the remark this answers is not in the document]
Comment (rarajes2) [reply docs/samples/calling/index.html:133]: Do we need to group outgoing and incoming call and put the call-object and user session in the bottom for both of them to share?

File: docs/samples/calling/index.html (hunk 4 above)
Comment (rarajes2) [reply docs/samples/calling/index.html:165]: Done

File: docs/samples/calling/app.js (hunk 5 above)
Comment (rarajes2) [reply docs/samples/calling/app.js:98]: All the sections are collapsible now. Try clicking on the section headers like `Authentication`, `Media Streams`, `Call initialization` etc except for the first one `Calling SDK`

File: docs/samples/calling/index.html (hunk 7 above)
Comment (rarajes2) [reply docs/samples/calling/index.html:131]: Done

File: docs/samples/calling/index.html [hunk 10]
```diff
@@ -9,495 +9,518 @@
...
-          <fieldset>
-            <legend>Call Quality</legend>
-            <div class="umv">
-              <pre id="call-quality-metrics">Call quality Metrics</pre>
-              <button id="get-call-quality" type="button" onclick="getCallQuality()" class="btn-code">Call
-                Quality</button>
+          
+            <form id="call-create">
+              <fieldset>
+                <legend>Make a call</legend>
```
[the remark this answers is not in the document]
Comment (rarajes2) [reply docs/samples/calling/index.html:126]: Done

File: docs/samples/calling/index.html (hunk 9 above)
[the remark this answers is not in the document]
Comment (rarajes2) [reply docs/samples/calling/index.html:133]: Done

File: packages/webex/src/calling.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (rarajes2) [reply packages/webex/src/calling.js]: Removed as `callingClient.mediaEngine.Media` worked.

File: docs/samples/calling/app.js (hunk 6 above)
Comment (rarajes2) [reply docs/samples/calling/app.js]: `callingClient.mediaEngine.Media` worked. Thanks

File: docs/samples/calling/index.html [hunk 11]
```diff
@@ -9,495 +9,515 @@
...
+            <div id="incomingsection">
+              <fieldset>
+                <legend>Incoming Call</legend>
+                <div class="u-mv">
+                  <pre id="incoming-call"> No Incoming Calls</pre>
+                  <button onclick="answer()" disabled="" id="answer" class="btn--green">Answer</button>
+                  <button onclick="endCall()" disabled="" id="end" class="btn--red">End Call</button>
+
+                  <pre id="call-object">Call-object</pre>
+                  <pre id="user_session">User-Session-Object</pre>
```
Comment (Kesari3008) [inline docs/samples/calling/index.html:145]: Bring them outside incoming call section.

File: docs/samples/calling/index.html [hunk 12]
```diff
@@ -9,495 +9,515 @@
...
-              <pre id="call-quality-metrics">Call quality Metrics</pre>
-              <button id="get-call-quality" type="button" onclick="getCallQuality()" class="btn-code">Call
-                Quality</button>
+          
+            <form id="call-create">
+              <fieldset>
+                <legend>Outgoing Call</legend>
+                <div class="u-mv">
+                  <input id="destination" name="destination" placeholder="Destination" type="string" style="display: block; margin-bottom: 0.5rem;">
+                  <button id="create-call-action" type="submit" class="btn-code btn--green">Make Call</button>
```
Comment (Kesari3008) [inline docs/samples/calling/index.html:129]: Once we make an outbound call, this button should also get disabled until we disconnect.

File: docs/samples/calling/index.html [hunk 13]
```diff
@@ -9,495 +9,515 @@
...
+            </form>
+
+            <!-- calling / incoming -->
+            <div id="incomingsection">
+              <fieldset>
+                <legend>Incoming Call</legend>
+                <div class="u-mv">
+                  <pre id="incoming-call"> No Incoming Calls</pre>
+                  <button onclick="answer()" disabled="" id="answer" class="btn--green">Answer</button>
+                  <button onclick="endCall()" disabled="" id="end" class="btn--red">End Call</button>
```
Comment (Kesari3008) [inline docs/samples/calling/index.html:142]: This button never gets disabled once becomes active. 
If the received incoming call is rejected then both the buttons remain active even after disconnect.

Correct these 2 behaviors

File: docs/samples/calling/index.html [hunk 14]
```diff
@@ -9,495 +9,515 @@
...
-        <!-- TODO: Once we have track changes in places and logic to update devices, below section will be added back -->
-        <!-- meeting-controls / media-devices
-        <fieldset id="control-devices">
-          <legend>Media Devices</legend>
-          <div>
-            <p class="note">NOTE: Choose your sending and receiving options for the meeting.</p>
+            <fieldset>
+              <legend>Call Controls</legend>
+              <div style="margin-bottom: 1rem;">
+                <input onclick="muteUnmute()" type="button" value="Mute" class="btn-code" id="mute_button"></input>
```
Comment (Kesari3008) [inline docs/samples/calling/index.html:154]: Mute/Unmute issue still persists, please check

**Review by Kesari3008 — CHANGES_REQUESTED** [review]
Found few issues while testing, please fix those and code should be good for approval

File: docs/samples/calling/index.html (hunk 11 above)
Comment (rarajes2) [reply docs/samples/calling/index.html:145]: Done

File: docs/samples/calling/index.html (hunk 12 above)
Comment (rarajes2) [reply docs/samples/calling/index.html:129]: Done

File: docs/samples/calling/index.html (hunk 13 above)
Comment (rarajes2) [reply docs/samples/calling/index.html:142]: Fixed

File: docs/samples/calling/index.html (hunk 14 above)
Comment (rarajes2) [reply docs/samples/calling/index.html:154]: Fixed

**Review by karthikraji2020 — APPROVED** [review]
Code Changes LGTM

**Review by Kesari3008 — APPROVED** [review]

**Pull request merged**